base/pkg/dir.jl

```diff
@@ -35,15 +35,20 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
     end
     dir = path()
     info("Initializing package repository $dir")
+    temp_dir = joinpath(dir, "TEMP")
     metadata_dir = joinpath(dir, "METADATA")
+    #Check if TEMP dir exists from previous failed init
+    if isdir(temp_dir)
```

Comment:
Just a thought, why not use base.mktempdir() and then you won't need to do this test.